packages/Radio/src/components/Group/Group.js

```diff
@@ -0,0 +1,69 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { ShirtSizes } from "@paprika/helpers/lib/customPropTypes";
+
+const propTypes = {
+  /** aria-labelledby prop on the containing group element */
+  a11yLabelledByText: PropTypes.string,
+
+  /** Can deselect any radio */
+  canDeselect: PropTypes.bool,
+
+  /** Function used to evaluate which radio is selected by default. */
+  defaultCheck: PropTypes.func,
+
+  /** The individual radio itemss. */
+  children: PropTypes.node,
+
+  /** Are all radios disabled */
+  isDisabled: PropTypes.bool,
+
+  /** On change of radio selection. */
+  onChange: PropTypes.func.isRequired,
+
+  /** The size for all radio components. */
+  size: PropTypes.oneOf(ShirtSizes.DEFAULT),
+};
+
+const defaultProps = {
+  a11yLabelledByText: "",
+  canDeselect: false,
+  children: null,
+  defaultCheck: () => {},
+  isDisabled: false,
+  size: ShirtSizes.MEDIUM,
+};
+
+function Group(props) {
+  const { a11yLabelledByText, canDeselect, children, defaultCheck, ...moreGroupProps } = props;
+  const defaultCheckedId = React.Children.toArray(children).find(defaultCheck).props.value.id;
+  const [checkedId, setCheckedId] = React.useState(defaultCheckedId || null);
+  const deselectableId = id => (checkedId === id ? null : id);
+  const handleRadioClick = id => {
+    setCheckedId(canDeselect ? deselectableId(id) : id);
+  };
+
+  return (
+    <div role="radiogroup" aria-labelledby={a11yLabelledByText} data-pka-anchor="radio.group">
+      {React.Children.map(children, (child, index) => {
+        if (child && child.type && child.type.displayName === "Radio") {
+          const childKey = { key: `Radio${index}` };
+          return React.cloneElement(child, {
```

Comment:
I suggest reviewing this to see if we can migrate to a context-based structure instead, see https://github.com/acl-services/paprika/pull/134/commits/f5063084fb5c90ab0477b913fc0ac8bd7b759b04 for an example. `cloneElement` is the least composition-friendly approach.

Reply:
Ok I'll look into changing to use this approach in a follow up PR, but I agree would be better.